packages/contracts-bedrock/tasks/genesis-l2.ts

```diff
@@ -166,14 +166,14 @@ const replaceImmutables = async (
       }
 
       // Ensure that the value being sliced out is 0
-      const val = hexDataSlice(
-        deployedBytecode,
-        offset.start,
-        offset.start + offset.length
-      )
-      if (!BigNumber.from(val).eq(0)) {
-        throw new Error(`Unexpected value in immutable bytecode ${val}`)
-      }
+      // const val = hexDataSlice(
```

Comment:
Is this causing a bug? If so, something is off

Reply:
Yes, forgot, this doesn't work in Hive for some reason. The contents are non-zero.

Reply:
I will need to investigate this, it would lead me to believe there is a bug in the way immutables are handled then. Weird that it works for non hive deployments..

Reply:
It makes me think the artifacts are out of date

Reply:
Fixed, problem with deploy/config order, thanks for your help yesterday :+1:

Reply:
Unresolving this. It's happening in devnet too.

Reply:
Will rebuild and try again, this is weird

Reply:
Still happens in devnet with a rebuild.
Getting:
```
An unexpected error occurred:

Error: Unexpected value in immutable bytecode, contract L2ToL1MessagePasser ast 22099 key MAJOR_VERSION path contracts/L2/L2ToL1MessagePasser.sol >  id 2517 >inherit contracts/universal/Semver.sol id 22158 >  id 22157 >  id 22099: 0x617a5ac1cc7eba2d6f37570a0135afa950d8bb77cdd35f0d0b4e85a16f906000
    at replaceImmutables (/home/proto/projects/optimism/packages/contracts-bedrock/tasks/genesis-l2.ts:181:15)
    at async SimpleTaskDefinition.action (/home/proto/projects/optimism/packages/contracts-bedrock/tasks/genesis-l2.ts:467:11)
    at async Environment._runTaskDefinition (/home/proto/projects/optimism/node_modules/hardhat/src/internal/core/runtime-environment.ts:219:14)
    at async Environment.run (/home/proto/projects/optimism/node_modules/hardhat/src/internal/core/runtime-environment.ts:131:14)
    at async main (/home/proto/projects/optimism/node_modules/hardhat/src/internal/cli/cli.ts:218:5)
make: *** [Makefile:59: devnet-up] Error 1

```
(added ast info to the error log)
So it looks like the MAJOR_VERSION immutable var is not zero in the bytecode.
And I'm surprised it has this large random looking string, and not a version number like the name indicates.

Reply:
Going to clear all caches and rebuild, maybe some bytecode is not correct

Reply:
deleting caches and rebuilding worked, oh well